### ISSUE TITLE
ragged.boolean_mask() gives a different result when executed in a model

### ISSUE THREAD

**Issue by surGeonGG** [tensorflow/tensorflow#41917]
```
mask = tf.ones((1, 100, 300), dtype=bool)

data = tf.ragged.boolean_mask(
    data=tf.ones((1,100,300,12)),
    mask=mask
).to_tensor()

print(data)
```

This gives me `tf.Tensor(......, shape=(1, 100, 300, 12), dtype=float32)` when executed normally, but when it is executed in a model it gives me `Tensor("model/attention/RaggedToTensor/RaggedTensorToTensor:0", shape=(None, None, None, 12), dtype=float32)`.

Why is this? How do I get it to return the correctly masked tensor in a model?

**Comment by Saduf2019** [comment]
@surGeonGG 
We see that you have not filled the issue template, please fill the issue template which provides us with the tf version, stand alone code the replicate the issue or is possible share a colab gist with the error faced.

**Comment by surGeonGG** [comment]
Replicated here

https://colab.research.google.com/drive/1HaOtA209X062l7OHn3JPtI2O_uGEcc8O?usp=sharing

**Comment by Saduf2019** [comment]
@surGeonGG 
if Dense is first layer then you have to set shape/size of input data. You can see it even in example in[ Dense documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense).

Please refer to below links for reference:
#35464 #33422 #23748 [link](https://stackoverflow.com/questions/56918388/error-valueerror-the-last-dimension-of-the-inputs-to-dense-should-be-defined)

**Comment by surGeonGG** [comment]
I'm confused. Dense is not the first layer. The first layer is an lstm. The shape information becomes (none, none, none) after I call...

` mask = tf.ones((1, 200, 12), dtype=bool)
    x = tf.ragged.boolean_mask(
        data=x,
        mask=mask
    ).to_tensor()`

**Comment by surGeonGG** [comment]
Figured it out! I simply use `x = tf.ensure_shape(x, (*shape here*))` afterwards to make sure the shape is correct.

Thanks for the help.
